src/path_manager.c

```diff
@@ -85,106 +85,145 @@ static bool validate_attr_len(size_t actual, size_t expected)
+                        MPTCP_GET_NL_ATTR(data, len, attrs->token);
+                        break;
+                case MPTCP_ATTR_LOC_ID:
+                        MPTCP_GET_NL_ATTR(data, len, attrs->laddr_id);
+                        break;
+                case MPTCP_ATTR_REM_ID:
+                        MPTCP_GET_NL_ATTR(data, len, attrs->raddr_id);
                         break;
                 case MPTCP_ATTR_SADDR4:
-                        MPTCP_GET_NL_ATTR(data, len, laddr4);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->laddr4);
                         break;
                 case MPTCP_ATTR_SADDR6:
-                        MPTCP_GET_NL_ATTR(data, len, laddr6);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->laddr6);
                         break;
                 case MPTCP_ATTR_SPORT:
-                        MPTCP_GET_NL_ATTR(data, len, local_port);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->local_port);
                         break;
                 case MPTCP_ATTR_DADDR4:
-                        MPTCP_GET_NL_ATTR(data, len, raddr4);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->raddr4);
                         break;
                 case MPTCP_ATTR_DADDR6:
-                        MPTCP_GET_NL_ATTR(data, len, raddr6);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->raddr6);
                         break;
                 case MPTCP_ATTR_DPORT:
-                        MPTCP_GET_NL_ATTR(data, len, remote_port);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->remote_port);
                         break;
-                case MPTCP_ATTR_FAMILY:
-                case MPTCP_ATTR_LOC_ID:
-                case MPTCP_ATTR_REM_ID:
                 case MPTCP_ATTR_BACKUP:
+                        MPTCP_GET_NL_ATTR(data, len, attrs->backup);
+                        break;
+                case MPTCP_ATTR_ERROR:
+                        MPTCP_GET_NL_ATTR(data, len, attrs->error);
+                        break;
                 case MPTCP_ATTR_IF_IDX:
-                        // Unused and ignored, at least for now.
+                        MPTCP_GET_NL_ATTR(data, len, attrs->index);
                         break;
-#ifdef MPTCPD_ENABLE_PM_NAME
-                case MPTCP_ATTR_PATH_MANAGER:
-                        pm_name = get_pm_name(data, len);
+                case MPTCP_ATTR_FAMILY:
+                case MPTCP_ATTR_FLAGS:
+                case MPTCP_ATTR_TIMEOUT:
+                case MPTCP_ATTR_RESET_REASON:
+                case MPTCP_ATTR_RESET_FLAGS:
+                        // Unused and ignored, at least for now.
                         break;
-#endif  // MPTCPD_ENABLE_PM_NAME
                 default:
-                        l_warn("Unknown MPTCP_EVENT_CREATED "
-                               "attribute: %d",
-                               type);
+                        l_warn("Unknown MPTCP genl attribute: %d", type);
```

Comment:
Should this be a warning or just informational? No harm in running on a newer kernel that might populate additional attributes.

Reply:
Informational is better.  I changed it accordingly.